Source/DafnyRuntime/DafnyRuntime.js

```diff
@@ -20,7 +20,7 @@ let _dafny = (function() {
       return a === b;
     } else if (BigNumber.isBigNumber(a)) {
       return a.isEqualTo(b);
-    } else if (a._tname !== undefined) {
+    } else if (a._tname !== undefined || (Array.isArray(a) && a.constructor.name == "Array")) {
```

Comment:
Also fixing a soundness bug?

Reply:
It had led to a crash at run time for the JavaScript target. I reported it as https://github.com/dafny-lang/dafny/issues/3207.